nixos/modules/services/hardware/auto-epp.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.auto-epp;
+  format = pkgs.formats.ini {};
+
+  inherit (lib) mkOption types;
+in {
+  options = {
+
+   services.auto-epp = {
+     enable = lib.mkEnableOption (lib.mdDoc "auto-epp for amd active pstate");
+
+     package = lib.mkPackageOptionMD pkgs "auto-epp" {};
+
+     settings = mkOption {
+       type = types.submodule {
+         freeformType = format.type;
+         options = {
+           Settings = {
+             epp_state_for_AC = mkOption {
+               type = types.str;
+               default = "power";
+               description = lib.mdDoc ''
+                 energy_performance_preference when on battery
+
+                   ::: {.note}
+                   See available epp states by running:
+                   {command}`cat /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_available_preferences`
+                   :::
+               '';
+             };
+
+             epp_state_for_BAT = mkOption {
+               type = types.str;
+               default = "balance_performance";
+               description = lib.mdDoc ''
+                 `energy_performance_preference` when plugged in
+
+                   ::: {.note}
+                   See available epp states by running:
+                   {command}`cat /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_available_preferences`
+                   :::
+               '';
+             };
+           };
+         };
+       };
+       default = {};
+       description = "Settings for auto-epp application";
+     };
+
+   };
+
+  };
```

Comment:
nit: fixed some odd-even indentation issues, added a missing `mdDoc` and removed some newlines
```suggestion
  options = {
    services.auto-epp = {
      enable = lib.mkEnableOption (lib.mdDoc "auto-epp for amd active pstate");

      package = lib.mkPackageOptionMD pkgs "auto-epp" {};

      settings = mkOption {
        type = types.submodule {
          freeformType = format.type;
          options = {
            Settings = {
              epp_state_for_AC = mkOption {
                type = types.str;
                default = "power";
                description = lib.mdDoc ''
                  energy_performance_preference when on battery

                  ::: {.note}
                  See available epp states by running:
                  {command}`cat /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_available_preferences`
                  :::
                '';
              };

              epp_state_for_BAT = mkOption {
                type = types.str;
                default = "balance_performance";
                description = lib.mdDoc ''
                  `energy_performance_preference` when plugged in

                  ::: {.note}
                  See available epp states by running:
                  {command}`cat /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_available_preferences`
                  :::
                '';
              };
            };
          };
        };
        default = {};
        description = lib.mdDoc "Settings for the auto-epp application";
      };
    };
  };
```

Reply:
any linter i can use that follows nixpkgs'  requirements ?

Reply:
While there are linters, I don't think this comment is a requirement for the most part. I know some people use [`nixpkgs-fmt`](https://nix-community.github.io/nixpkgs-fmt/)